pkg/destroy/gcp/filestore.go

```diff
@@ -0,0 +1,96 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/api/file/v1"
+
+	gcpconsts "github.com/openshift/installer/pkg/constants/gcp"
+)
+
+func (o *ClusterUninstaller) filestoreParentPath() string {
+	return fmt.Sprintf("projects/%s/locations/-", o.ProjectID)
+}
+
+func (o *ClusterUninstaller) clusterFilestoreLabelFilter() string {
+	return fmt.Sprintf("labels.%s = \"owned\"", fmt.Sprintf(gcpconsts.ClusterIDLabelFmt, o.ClusterID))
+}
+
+func (o *ClusterUninstaller) listFilestores(ctx context.Context) ([]cloudResource, error) {
+	o.Logger.Debug("Listing filestores")
+	result := []cloudResource{}
+
+	ctx, cancel := context.WithTimeout(ctx, defaultTimeout)
+	defer cancel()
+	instSvc := file.NewProjectsLocationsInstancesService(o.fileSvc)
+	lCall := instSvc.List(o.filestoreParentPath()).Context(ctx).Filter(o.clusterFilestoreLabelFilter())
+
+	nextPageToken := "pageToken"
+	for nextPageToken != "" {
+		instances, err := lCall.Do()
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving filestore instances: %w", err)
+		}
+
+		for _, activeInstance := range instances.Instances {
+			if len(activeInstance.FileShares) == 0 {
+				// skip multi-share instances
```

Comment:
OpenShift doesn't support creation of multishare Filestore instances (yet). They allow for allocation of multiple volumes per one instance and the idea behind this bit of code was that if we encounter such an instance, we should simply ignore it since it's most likely being used by multiple clusters and it's safer to leak the resource than to lose data of the other Filestore users.

Documentation for the reference: https://cloud.google.com/filestore/docs/multishares

Reply:
So I discussed the above with @jsafrane and the conclusion was that the reasoning I provided was not very clever, so I'll remove that check. It is a no-op, but in the case we add support for multi-shares, the "owned" label should mean the cluster owns the resource, so it should get removed along with the cluster.